event_model/schemas/event_descriptor.json

```diff
@@ -1,150 +1,142 @@
-                },
-                "type": "object",
-                "description": "This describes the data in the Event Documents.",
-                "title": "data_keys"
-            },
-            "uid": {
-                "type": "string",
-                "description": "Globally unique ID for this event descriptor.",
-                "title": "uid"
-            },
-            "run_start": {
-                "type": "string",
-                "description": "Globally unique ID of this run's 'start' document."
+            "shape": {
+               "title": "Shape",
+               "description": "The shape of the data.  Empty list indicates scalar data.",
+               "type": "array",
+               "items": {
+                  "type": "integer"
+               }
             },
-            "time": {
-                "type": "number",
-                "description": "Creation time of the document as unix epoch time."
-            },
-            "hints": {
-                "$ref": "#/definitions/object_hints"
-            },
-            "object_keys": {
-                "type": "object",
-                "description": "Maps a Device/Signal name to the names of the entries it produces in data_keys."
+            "source": {
+               "title": "Source",
+               "description": "The source (ex piece of hardware) of the data.",
+               "type": "string"
+            }
+         },
+         "required": [
+            "dtype",
+            "shape",
+            "source"
+         ]
+      },
+      "Configuration": {
+         "title": "Configuration",
+         "type": "object",
+         "properties": {
+            "data": {
+               "title": "Data",
+               "description": "The actual measurement data",
+               "type": "object"
             },
-            "name": {
-                "type": "string",
-                "description": "A human-friendly name for this data stream, such as 'primary' or 'baseline'."
+            "timestamps": {
+               "title": "Timestamps",
+               "description": "The timestamps of the individual measurement data",
+               "type": "object"
             },
-            "configuration": {
-                "additionalProperties": {
-                    "$ref": "#/definitions/configuration"
-                },
-                "type": "object",
-                "description": "Readings of configurational fields necessary for interpreting data in the Events."
+            "data_keys": {
+               "title": "Data Keys",
+               "description": "This describes the data stored alongside it in this configuration object.",
+               "type": "object",
+               "additionalProperties": {
+                  "$ref": "#/definitions/DataKey"
+               }
             }
-        },
-        "patternProperties": {
-            "^([^./]+)$": {"$ref": "#/definitions/data_type"}},
```

Comment:
From my [reading](https://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties)  this will have any string with no `./` becoming a data_type. Am I right about this @danielballan